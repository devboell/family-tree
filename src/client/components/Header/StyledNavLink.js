import styled from 'styled-components'
import { NavLink } from 'react-router-dom'


export const activeClassName = 'nav-item-active'

const StyledNavLink = styled(NavLink).attrs({
  activeClassName,
})`
  text-decoration: none;
  display: inline-block;
  > h3 {
    margin-left: 10px;
    margin-bottom: 10px;
    color: black;
    border-bottom: solid transparent 1px;
  }

  &.${activeClassName} {
    > h3 {
      border-bottom: solid black 1px;
    }
  }
`

export default StyledNavLink
