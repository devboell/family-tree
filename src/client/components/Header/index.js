import React from 'react'

import { NavLink } from 'react-router-dom'

const Header = () =>
  <div>
    <NavLink to="/">Tree</NavLink>
    <NavLink to="/editor">Editor</NavLink>
  </div>


export default Header
