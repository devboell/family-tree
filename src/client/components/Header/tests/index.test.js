import React from 'react'
import Header from '../index'

it('Header component, it renders without crashing', () => {
  expect(shallow(<Header />)).toMatchSnapshot()
})
