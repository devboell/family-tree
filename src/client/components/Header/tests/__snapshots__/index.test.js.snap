// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component, it renders without crashing 1`] = `
<div>
  <Styled(NavLink)
    exact={true}
    to="/"
  >
    <h3>
      Tree
    </h3>
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/editor"
  >
    <h3>
      Editor
    </h3>
  </Styled(NavLink)>
</div>
`;
